### ISSUE TITLE
As a user, I want to find members based on partial name matches so that I can find all variations of a name.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/mep-django#363]
## notes for testing
- searching members should now return results when part of the name matches; search term must be at least 3 characters long to trigger this behavior
- exact matches should be more relevant than partial matches (e.g., if you search "stein" Gertrude and Leo should be listed first before all the other *stein names)
- partial matching should ignore case an accents
- partial matching should work anywhere within the name, on both sort name and name
- partial matching should work with more than search term

Note that the partial matching ngram is currently configured to generate ngram sizes 3-8, so partial matches longer than 8 characters won't have this behavior.

## dev notes
- configure a new solr field type with n-gram indexing (3 minimum)
- configure a new name field that uses the new field type
- copy from sort name and name to the new field
- include the new field in the solr name query field, but with a lower boost than the main names

**Comment by rlskoeser** [comment]
It looks like the right way to do this is to configure an [N-gram filter](https://lucene.apache.org/solr/guide/6_6/filter-descriptions.html#FilterDescriptions-N-GramFilter) for indexing. We'll need to choose a minimum number of characters.

**Comment by thatbudakguy** [comment]
this is awesome - and potentially super relevant for DIRECT. good find!!

**Comment by clmahoney** [comment]
Desktop: 
- YES, if I type in just 3 letters I start getting results - typed in "Jam" and the James' all appeared 
- NO, I don't think exact matches are giving me the most relevant first. I tried "stein" and Gertrude WAS the first, but then Leo was the third. For some reason "Stein, Irwin" was the second. Same with James Joyce: I typed in "Joyce" and JJ was second, not first. For some reason "Linden, Joyce" was first. 
- YES, partial matching should ignore case an accents - I typed in "Rene" and I got every instance of that combination of letters, including René and Renée. 
- YES, partial matching does work anywhere within the name, on both sort name and name - so for the "rene" example, I got Irene, but I also got sort name "Mayran, Camille" because her full name has "rene" in it too (listed in the bio box)

- partial matching should work with more than search term --> can you clarify this?

**Comment by rlskoeser** [comment]
@clmahoney I don't understand your comment about exact matches being made more relevant - Gertrude, Leo, and Erwin are all exact matches for "stein" and should be listed before all the other steins (Birstein, Goldstein, etc).

What I meant about more than one search term is that the partial search logic isn't limited to a single search term  - e.g. "gert stei" should find Gertrude Stein.

**Comment by clmahoney** [comment]
@rlskoeser oh OK, then I was misunderstanding that and the partial search works well to isolate the most important steins!

**Comment by clmahoney** [comment]
unfortunately NO, "gert stei" does not yield Gertrude Stein

**Comment by clmahoney** [comment]
However, "jam joy" DOES yield James Joyce!

**Comment by clmahoney** [comment]
@rlskoeser ah yes I tried "gert ste" again and there she is, Gertrude Stein!!

**Comment by clmahoney** [comment]
Mobile: 
- YES, if I type in just 3 letters I start getting results - typed in "pri" and everything from the many Price names to Cirpiani, to Princess Bassiano appeared. Also tried "Bido" and Bidot and Bidoire appeared.  
- YES, exact matches give me the most relevant first. "Stein" gave me the top 3 Steins (Gertrude, Irwin, Leo) before showing me names that ended with -stein
- YES, partial matching ignores accents - I typed in "Rene" and I got every instance of that combination of letters, including René and Renée.
- YES, partial matching does work anywhere within the name, on both sort name and name - so for the "rene" example, I got Irene, but I also got sort name "Mayran, Camille" because her full name has "rene" in it too 
- YES, partial matching works with more than one search term: jam joy still yields James Joyce!

**Comment by clmahoney** [comment]
Tablet: 
- YES, if I type in just 3 letters I start getting results - typed in "rap" and got all 3 Raphael's and also Grapard and Rapael 
- YES, exact matches give me the most relevant first. "Stein" gave me the top 3 Steins (Gertrude, Irwin, Leo) before showing me names that ended with -stein
- YES, partial matching ignores accents - I typed in "Rene" and I got every instance of that combination of letters, including René and Renée.
- YES, partial matching does work anywhere within the name, on both sort name and name - so for the "rene" example, I got Irene, but I also got sort name "Mayran, Camille" because her full name has "rene" in it too
- YES, partial matching works with more than one search term: Gert ste yields Gertrude Stein!
